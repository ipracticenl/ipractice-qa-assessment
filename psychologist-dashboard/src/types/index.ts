// Base types
export interface User {
  id: number;
  name: string;
  type: "psychologist" | "client";
}

// Time range base type
export interface TimeRange {
  id: string;
  from: Date;
  to: Date;
}

// Client DTOs
export interface CreateClientDto {
  name: string;
  initialPsychologistIds: number[];
}

export interface ClientSummaryDto {
  id: number;
  name: string;
}

export interface ClientDetailsDto {
  id: number;
  name: string;
  assignedPsychologists: number[];
  bookedAppointments: Appointment[];
}

// Psychologist DTOs
export interface CreatePsychologistDto {
  name: string;
  initialClients: number[];
}

export interface PsychologistSummaryDto {
  id: number;
  name: string;
}

export interface PsychologistDetailsDto {
  id: number;
  name: string;
  assignedClients: number[];
  availableTimeSlots: AvailableTimeSlot[];
  bookedAppointments: BookedAppointment[];
}

// Booking DTOs
export interface CreateBookingDto {
  psychologistId: number;
  availableTimeSlotId: string;
  comment?: string;
}

export interface ClientDto {
  clientId: number;
}

export interface TimeSlotDto {
  from: Date;
  to: Date;
}

// Model types
export interface AvailableTimeSlot extends TimeRange {
  // Inherits id, from, to from TimeRange
}

export interface Appointment extends TimeRange {
  psychologistId: number;
  comment?: string;
}

export interface BookedAppointment extends TimeRange {
  clientId: number;
}

// Legacy types for backward compatibility
export interface Psychologist extends User {
  type: "psychologist";
  clients: string[];
}

export interface Client extends User {
  type: "client";
  psychologist?: string;
}

export interface Availability {
  id: string;
  psychologistId: number;
  startTime: Date;
  endTime: Date;
  isBooked: boolean;
  clientId?: string;
  clientName?: string;
}
