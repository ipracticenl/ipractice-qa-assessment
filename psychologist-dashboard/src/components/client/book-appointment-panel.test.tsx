import { render, screen, fireEvent } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import { BookAppointmentPanel } from "./book-appointment-panel";
import { AvailableTimeSlot, PsychologistSummaryDto } from "@/types";

// Mock data with local timezone dates
const mockPsychologists: PsychologistSummaryDto[] = [
  { id: 1, name: "Dr. John Smith" },
  { id: 2, name: "Dr. Jane Doe" },
];

const mockAvailableSlots: AvailableTimeSlot[] = [
  {
    id: "slot-1",
    from: new Date("2024-01-15T10:00:00"),
    to: new Date("2024-01-15T11:00:00"),
  },
  {
    id: "slot-2",
    from: new Date("2024-01-16T14:00:00"),
    to: new Date("2024-01-16T15:00:00"),
  },
];

describe("BookAppointmentPanel", () => {
  const mockSetShowBooking = vi.fn();
  const mockSetSelectedPsychologist = vi.fn();
  const mockResetBookingFlow = vi.fn();
  const mockHandleBookAppointment = vi.fn();

  describe("Snapshot Tests", () => {
    it("should match snapshot when showing psychologist selection", () => {
      // Given: The component is in psychologist selection mode
      const showBooking = false;
      const selectedPsychologistId = null;

      // When: The component is rendered in selection mode
      const { container } = render(
        <BookAppointmentPanel
          psychologists={mockPsychologists}
          availableSlots={mockAvailableSlots}
          showBooking={showBooking}
          setShowBooking={mockSetShowBooking}
          selectedPsychologistId={selectedPsychologistId}
          setSelectedPsychologist={mockSetSelectedPsychologist}
          resetBookingFlow={mockResetBookingFlow}
          handleBookAppointment={mockHandleBookAppointment}
        />
      );

      // Then: The component should match the snapshot
      expect(container).toMatchSnapshot();
    });

    it("should match snapshot when showing available slots", () => {
      // Given: The component is in booking mode with available slots
      const showBooking = true;
      const selectedPsychologistId = 1;

      // When: The component is rendered in booking mode
      const { container } = render(
        <BookAppointmentPanel
          psychologists={mockPsychologists}
          availableSlots={mockAvailableSlots}
          showBooking={showBooking}
          setShowBooking={mockSetShowBooking}
          selectedPsychologistId={selectedPsychologistId}
          setSelectedPsychologist={mockSetSelectedPsychologist}
          resetBookingFlow={mockResetBookingFlow}
          handleBookAppointment={mockHandleBookAppointment}
        />
      );

      // Then: The component should match the snapshot
      expect(container).toMatchSnapshot();
    });

    it("should match snapshot when no available slots exist", () => {
      // Given: The component is in booking mode with no available slots
      const showBooking = true;
      const selectedPsychologistId = 1;
      const emptySlots: AvailableTimeSlot[] = [];

      // When: The component is rendered with no available slots
      const { container } = render(
        <BookAppointmentPanel
          psychologists={mockPsychologists}
          availableSlots={emptySlots}
          showBooking={showBooking}
          setShowBooking={mockSetShowBooking}
          selectedPsychologistId={selectedPsychologistId}
          setSelectedPsychologist={mockSetSelectedPsychologist}
          resetBookingFlow={mockResetBookingFlow}
          handleBookAppointment={mockHandleBookAppointment}
        />
      );

      // Then: The component should match the snapshot
      expect(container).toMatchSnapshot();
    });
  });

  describe("Psychologist Selection Mode", () => {
    it("should display psychologist selection when showBooking is false", () => {
      // Given: The component is in selection mode
      const showBooking = false;

      // When: The component is rendered
      render(
        <BookAppointmentPanel
          psychologists={mockPsychologists}
          availableSlots={mockAvailableSlots}
          showBooking={showBooking}
          setShowBooking={mockSetShowBooking}
          selectedPsychologistId={null}
          setSelectedPsychologist={mockSetSelectedPsychologist}
          resetBookingFlow={mockResetBookingFlow}
          handleBookAppointment={mockHandleBookAppointment}
        />
      );

      // Then: The psychologist selector should be displayed
      expect(screen.getByText("Select Psychologist")).toBeInTheDocument();
      expect(screen.getByText("Choose a psychologist")).toBeInTheDocument();
    });

    it("should disable view available times button when no psychologist is selected", () => {
      // Given: No psychologist is selected
      const showBooking = false;
      const selectedPsychologistId = null;

      // When: The component is rendered
      render(
        <BookAppointmentPanel
          psychologists={mockPsychologists}
          availableSlots={mockAvailableSlots}
          showBooking={showBooking}
          setShowBooking={mockSetShowBooking}
          selectedPsychologistId={selectedPsychologistId}
          setSelectedPsychologist={mockSetSelectedPsychologist}
          resetBookingFlow={mockResetBookingFlow}
          handleBookAppointment={mockHandleBookAppointment}
        />
      );

      // Then: The view available times button should be disabled
      const viewButton = screen.getByTestId("view-available-times-button");
      expect(viewButton).toBeDisabled();
    });

    it("should enable view available times button when psychologist is selected", () => {
      // Given: A psychologist is selected
      const showBooking = false;
      const selectedPsychologistId = 1;

      // When: The component is rendered
      render(
        <BookAppointmentPanel
          psychologists={mockPsychologists}
          availableSlots={mockAvailableSlots}
          showBooking={showBooking}
          setShowBooking={mockSetShowBooking}
          selectedPsychologistId={selectedPsychologistId}
          setSelectedPsychologist={mockSetSelectedPsychologist}
          resetBookingFlow={mockResetBookingFlow}
          handleBookAppointment={mockHandleBookAppointment}
        />
      );

      // Then: The view available times button should be enabled
      const viewButton = screen.getByTestId("view-available-times-button");
      expect(viewButton).not.toBeDisabled();
    });

    it("should call setShowBooking when view available times button is clicked", () => {
      // Given: A psychologist is selected
      const showBooking = false;
      const selectedPsychologistId = 1;

      // When: The component is rendered and view button is clicked
      render(
        <BookAppointmentPanel
          psychologists={mockPsychologists}
          availableSlots={mockAvailableSlots}
          showBooking={showBooking}
          setShowBooking={mockSetShowBooking}
          selectedPsychologistId={selectedPsychologistId}
          setSelectedPsychologist={mockSetSelectedPsychologist}
          resetBookingFlow={mockResetBookingFlow}
          handleBookAppointment={mockHandleBookAppointment}
        />
      );

      const viewButton = screen.getByTestId("view-available-times-button");
      fireEvent.click(viewButton);

      // Then: setShowBooking should be called with true
      expect(mockSetShowBooking).toHaveBeenCalledWith(true);
    });
  });

  describe("Available Slots Display Mode", () => {
    it("should display available slots when showBooking is true", () => {
      // Given: The component is in booking mode with available slots
      const showBooking = true;
      const selectedPsychologistId = 1;

      // When: The component is rendered
      render(
        <BookAppointmentPanel
          psychologists={mockPsychologists}
          availableSlots={mockAvailableSlots}
          showBooking={showBooking}
          setShowBooking={mockSetShowBooking}
          selectedPsychologistId={selectedPsychologistId}
          setSelectedPsychologist={mockSetSelectedPsychologist}
          resetBookingFlow={mockResetBookingFlow}
          handleBookAppointment={mockHandleBookAppointment}
        />
      );

      // Then: Available slots should be displayed
      expect(screen.getByTestId("available-slots-list")).toBeInTheDocument();
      expect(screen.getByTestId("available-slot-slot-1")).toBeInTheDocument();
      expect(screen.getByTestId("available-slot-slot-2")).toBeInTheDocument();
    });

    it("should display psychologist name in the header", () => {
      // Given: The component is in booking mode with a selected psychologist
      const showBooking = true;
      const selectedPsychologistId = 1;

      // When: The component is rendered
      render(
        <BookAppointmentPanel
          psychologists={mockPsychologists}
          availableSlots={mockAvailableSlots}
          showBooking={showBooking}
          setShowBooking={mockSetShowBooking}
          selectedPsychologistId={selectedPsychologistId}
          setSelectedPsychologist={mockSetSelectedPsychologist}
          resetBookingFlow={mockResetBookingFlow}
          handleBookAppointment={mockHandleBookAppointment}
        />
      );

      // Then: The psychologist name should be displayed in the header
      expect(
        screen.getByText("Available times for Dr. John Smith")
      ).toBeInTheDocument();
    });

    it.skip("should display slot information correctly", () => {
      // Given: The component is in booking mode with available slots
      const showBooking = true;
      const selectedPsychologistId = 1;

      // When: The component is rendered
      render(
        <BookAppointmentPanel
          psychologists={mockPsychologists}
          availableSlots={mockAvailableSlots}
          showBooking={showBooking}
          setShowBooking={mockSetShowBooking}
          selectedPsychologistId={selectedPsychologistId}
          setSelectedPsychologist={mockSetSelectedPsychologist}
          resetBookingFlow={mockResetBookingFlow}
          handleBookAppointment={mockHandleBookAppointment}
        />
      );

      // Then: Slot information should be displayed correctly
      expect(screen.getByText("Monday, 01/15/2024")).toBeInTheDocument();
      expect(screen.getByText("Tuesday, 01/16/2024")).toBeInTheDocument();
      // Check for time ranges with flexible matching
      expect(
        // @ts-expect-error - This is a test
        screen.getByText((_content, element) => {
          return (
            element?.textContent?.includes("10:00") &&
            element?.textContent?.includes("11:00")
          );
        })
      ).toBeInTheDocument();
      expect(
        // @ts-expect-error - This is a test
        screen.getByText((_content, element) => {
          return (
            element?.textContent?.includes("14:00") &&
            element?.textContent?.includes("15:00")
          );
        })
      ).toBeInTheDocument();
    });

    it("should display empty state when no slots are available", () => {
      // Given: The component is in booking mode with no available slots
      const showBooking = true;
      const selectedPsychologistId = 1;
      const emptySlots: AvailableTimeSlot[] = [];

      // When: The component is rendered
      render(
        <BookAppointmentPanel
          psychologists={mockPsychologists}
          availableSlots={emptySlots}
          showBooking={showBooking}
          setShowBooking={mockSetShowBooking}
          selectedPsychologistId={selectedPsychologistId}
          setSelectedPsychologist={mockSetSelectedPsychologist}
          resetBookingFlow={mockResetBookingFlow}
          handleBookAppointment={mockHandleBookAppointment}
        />
      );

      // Then: Empty state message should be displayed
      expect(screen.getByText("No available time slots")).toBeInTheDocument();
      expect(
        screen.getByText(
          "Please check back later or select a different psychologist"
        )
      ).toBeInTheDocument();
    });
  });

  describe("Booking Functionality", () => {
    it.skip("should call handleBookAppointment when book button is clicked", () => {
      // Given: The component is in booking mode with available slots
      const showBooking = true;
      const selectedPsychologistId = 1;

      // When: The component is rendered and book button is clicked
      render(
        <BookAppointmentPanel
          psychologists={mockPsychologists}
          availableSlots={mockAvailableSlots}
          showBooking={showBooking}
          setShowBooking={mockSetShowBooking}
          selectedPsychologistId={selectedPsychologistId}
          setSelectedPsychologist={mockSetSelectedPsychologist}
          resetBookingFlow={mockResetBookingFlow}
          handleBookAppointment={mockHandleBookAppointment}
        />
      );

      const bookButton = screen.getByTestId("book-slot-slot-1");
      fireEvent.click(bookButton);

      // Then: handleBookAppointment should be called with the correct slot ID
      expect(mockHandleBookAppointment).toHaveBeenCalledWith("slot-1");
    });

    it("should have correct aria-label for book buttons", () => {
      // Given: The component is in booking mode with available slots
      const showBooking = true;
      const selectedPsychologistId = 1;

      // When: The component is rendered
      render(
        <BookAppointmentPanel
          psychologists={mockPsychologists}
          availableSlots={mockAvailableSlots}
          showBooking={showBooking}
          setShowBooking={mockSetShowBooking}
          selectedPsychologistId={selectedPsychologistId}
          setSelectedPsychologist={mockSetSelectedPsychologist}
          resetBookingFlow={mockResetBookingFlow}
          handleBookAppointment={mockHandleBookAppointment}
        />
      );

      // Then: Book buttons should have correct aria-labels
      const bookButton1 = screen.getByTestId("book-slot-slot-1");
      expect(bookButton1).toHaveAttribute("aria-label");
      expect(bookButton1.getAttribute("aria-label")).toContain(
        "Book appointment on Monday, 01/15/2024"
      );

      const bookButton2 = screen.getByTestId("book-slot-slot-2");
      expect(bookButton2).toHaveAttribute("aria-label");
      expect(bookButton2.getAttribute("aria-label")).toContain(
        "Book appointment on Tuesday, 01/16/2024"
      );
    });
  });

  describe("Navigation", () => {
    it("should call resetBookingFlow when back button is clicked", () => {
      // Given: The component is in booking mode
      const showBooking = true;
      const selectedPsychologistId = 1;

      // When: The component is rendered and back button is clicked
      render(
        <BookAppointmentPanel
          psychologists={mockPsychologists}
          availableSlots={mockAvailableSlots}
          showBooking={showBooking}
          setShowBooking={mockSetShowBooking}
          selectedPsychologistId={selectedPsychologistId}
          setSelectedPsychologist={mockSetSelectedPsychologist}
          resetBookingFlow={mockResetBookingFlow}
          handleBookAppointment={mockHandleBookAppointment}
        />
      );

      const backButton = screen.getByTestId("back-to-psychologist-selection");
      fireEvent.click(backButton);

      // Then: resetBookingFlow should be called
      expect(mockResetBookingFlow).toHaveBeenCalled();
    });
  });

  describe("Component Structure", () => {
    it("should render the main panel with correct test ID", () => {
      // Given: The component is in selection mode
      const showBooking = false;

      // When: The component is rendered
      render(
        <BookAppointmentPanel
          psychologists={mockPsychologists}
          availableSlots={mockAvailableSlots}
          showBooking={showBooking}
          setShowBooking={mockSetShowBooking}
          selectedPsychologistId={null}
          setSelectedPsychologist={mockSetSelectedPsychologist}
          resetBookingFlow={mockResetBookingFlow}
          handleBookAppointment={mockHandleBookAppointment}
        />
      );

      // Then: The main panel should be rendered with correct test ID
      expect(screen.getByTestId("book-appointment-panel")).toBeInTheDocument();
    });

    it("should display the correct title", () => {
      // Given: The component is in selection mode
      const showBooking = false;

      // When: The component is rendered
      render(
        <BookAppointmentPanel
          psychologists={mockPsychologists}
          availableSlots={mockAvailableSlots}
          showBooking={showBooking}
          setShowBooking={mockSetShowBooking}
          selectedPsychologistId={null}
          setSelectedPsychologist={mockSetSelectedPsychologist}
          resetBookingFlow={mockResetBookingFlow}
          handleBookAppointment={mockHandleBookAppointment}
        />
      );

      // Then: The title should be displayed
      expect(screen.getByText("Book New Appointment")).toBeInTheDocument();
    });
  });
});
