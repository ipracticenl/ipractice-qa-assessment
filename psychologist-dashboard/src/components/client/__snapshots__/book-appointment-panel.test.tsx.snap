// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BookAppointmentPanel > Snapshot Tests > should match snapshot when no available slots exist 1`] = `
<div>
  <div
    class="rounded-lg border bg-card text-card-foreground shadow-sm"
    data-testid="book-appointment-panel"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
    >
      <div
        class="text-2xl font-semibold leading-none tracking-tight flex items-center gap-2"
      >
        <svg
          class="lucide lucide-calendar w-5 h-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 2v4"
          />
          <path
            d="M16 2v4"
          />
          <rect
            height="18"
            rx="2"
            width="18"
            x="3"
            y="4"
          />
          <path
            d="M3 10h18"
          />
        </svg>
        Book New Appointment
      </div>
    </div>
    <div
      class="p-6 pt-0 space-y-4"
    >
      <div>
        <div
          class="flex flex-col sm:flex-row justify-between items-start sm:items-center mb-4 gap-2"
        >
          <h3
            class="font-medium text-lg"
          >
            Available times for
             
            Dr. John Smith
          </h3>
          <button
            class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3 min-h-[36px] touch-manipulation bg-transparent"
            data-testid="back-to-psychologist-selection"
          >
            <svg
              class="lucide lucide-arrow-left w-4 h-4 mr-1"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m12 19-7-7 7-7"
              />
              <path
                d="M19 12H5"
              />
            </svg>
            Back
          </button>
        </div>
        <div
          class="space-y-2 max-h-80 overflow-y-auto"
          data-testid="available-slots-list"
        >
          <div
            class="text-center py-8 text-gray-500"
          >
            <svg
              class="lucide lucide-clock w-12 h-12 mx-auto mb-2 text-gray-300"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <polyline
                points="12 6 12 12 16 14"
              />
            </svg>
            <p>
              No available time slots
            </p>
            <p
              class="text-sm"
            >
              Please check back later or select a different psychologist
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`BookAppointmentPanel > Snapshot Tests > should match snapshot when showing available slots 1`] = `
<div>
  <div
    class="rounded-lg border bg-card text-card-foreground shadow-sm"
    data-testid="book-appointment-panel"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
    >
      <div
        class="text-2xl font-semibold leading-none tracking-tight flex items-center gap-2"
      >
        <svg
          class="lucide lucide-calendar w-5 h-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 2v4"
          />
          <path
            d="M16 2v4"
          />
          <rect
            height="18"
            rx="2"
            width="18"
            x="3"
            y="4"
          />
          <path
            d="M3 10h18"
          />
        </svg>
        Book New Appointment
      </div>
    </div>
    <div
      class="p-6 pt-0 space-y-4"
    >
      <div>
        <div
          class="flex flex-col sm:flex-row justify-between items-start sm:items-center mb-4 gap-2"
        >
          <h3
            class="font-medium text-lg"
          >
            Available times for
             
            Dr. John Smith
          </h3>
          <button
            class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3 min-h-[36px] touch-manipulation bg-transparent"
            data-testid="back-to-psychologist-selection"
          >
            <svg
              class="lucide lucide-arrow-left w-4 h-4 mr-1"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m12 19-7-7 7-7"
              />
              <path
                d="M19 12H5"
              />
            </svg>
            Back
          </button>
        </div>
        <div
          class="space-y-2 max-h-80 overflow-y-auto"
          data-testid="available-slots-list"
        >
          <div
            class="flex flex-col sm:flex-row justify-between items-start sm:items-center p-4 border rounded-md hover:bg-gray-50 gap-2"
            data-testid="available-slot-slot-1"
          >
            <div
              class="flex-1"
            >
              <div
                class="font-medium"
              >
                Monday, 01/15/2024
              </div>
              <div
                class="text-sm text-gray-600 flex items-center gap-1"
              >
                <svg
                  class="lucide lucide-clock w-3 h-3"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="10"
                  />
                  <polyline
                    points="12 6 12 12 16 14"
                  />
                </svg>
                10:00
                 -
                 
                11:00
              </div>
            </div>
            <button
              aria-label="Book appointment on Monday, 01/15/2024 at 10:00"
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground hover:bg-primary/90 h-9 rounded-md px-3 min-h-[36px] touch-manipulation w-full sm:w-auto"
              data-testid="book-slot-slot-1"
            >
              Book
            </button>
          </div>
          <div
            class="flex flex-col sm:flex-row justify-between items-start sm:items-center p-4 border rounded-md hover:bg-gray-50 gap-2"
            data-testid="available-slot-slot-2"
          >
            <div
              class="flex-1"
            >
              <div
                class="font-medium"
              >
                Tuesday, 01/16/2024
              </div>
              <div
                class="text-sm text-gray-600 flex items-center gap-1"
              >
                <svg
                  class="lucide lucide-clock w-3 h-3"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="10"
                  />
                  <polyline
                    points="12 6 12 12 16 14"
                  />
                </svg>
                14:00
                 -
                 
                15:00
              </div>
            </div>
            <button
              aria-label="Book appointment on Tuesday, 01/16/2024 at 14:00"
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground hover:bg-primary/90 h-9 rounded-md px-3 min-h-[36px] touch-manipulation w-full sm:w-auto"
              data-testid="book-slot-slot-2"
            >
              Book
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`BookAppointmentPanel > Snapshot Tests > should match snapshot when showing psychologist selection 1`] = `
<div>
  <div
    class="rounded-lg border bg-card text-card-foreground shadow-sm"
    data-testid="book-appointment-panel"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
    >
      <div
        class="text-2xl font-semibold leading-none tracking-tight flex items-center gap-2"
      >
        <svg
          class="lucide lucide-calendar w-5 h-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 2v4"
          />
          <path
            d="M16 2v4"
          />
          <rect
            height="18"
            rx="2"
            width="18"
            x="3"
            y="4"
          />
          <path
            d="M3 10h18"
          />
        </svg>
        Book New Appointment
      </div>
    </div>
    <div
      class="p-6 pt-0 space-y-4"
    >
      <div>
        <label
          class="block text-sm font-medium mb-2"
          for="psychologist-selector"
        >
          Select Psychologist
        </label>
        <div
          class="flex flex-col sm:flex-row gap-2"
        >
          <button
            aria-autocomplete="none"
            aria-controls="radix-:r0:"
            aria-expanded="false"
            class="flex h-10 items-center justify-between rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background data-[placeholder]:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 w-full"
            data-placeholder=""
            data-state="closed"
            dir="ltr"
            role="combobox"
            type="button"
          >
            <span
              style="pointer-events: none;"
            >
              Choose a psychologist
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-down h-4 w-4 opacity-50"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m6 9 6 6 6-6"
              />
            </svg>
          </button>
          <div
            class="sr-only"
            id="psychologist-selector-help"
          >
            Select a psychologist to view their available appointment times
          </div>
          <button
            aria-describedby="view-times-help"
            class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 w-full min-h-[44px] touch-manipulation"
            data-testid="view-available-times-button"
            disabled=""
          >
            View Available Times
          </button>
        </div>
        <div
          class="sr-only"
          id="view-times-help"
        >
          View available appointment times for the selected psychologist
        </div>
      </div>
    </div>
  </div>
</div>
`;
